src/transformers/modeling_utils.py

```diff
@@ -2108,62 +2135,66 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         keep_in_fp32_modules = None
         use_keep_in_fp32_modules = False
 
+        # `variant` will be passed as suffix to weight names, if left None we can transform it to ""
+        variant_suffix = f".{variant}" if variant is not None else ""
+
         if pretrained_model_name_or_path is not None:
             pretrained_model_name_or_path = str(pretrained_model_name_or_path)
             is_local = os.path.isdir(pretrained_model_name_or_path)
             if is_local:
-                if from_tf and os.path.isfile(
-                    os.path.join(pretrained_model_name_or_path, subfolder, TF_WEIGHTS_NAME + ".index")
-                ):
+
+                def get_local_path(file_name):
+                    return os.path.join(pretrained_model_name_or_path, subfolder, file_name)
+
+                tf_weights_1_path = get_local_path(TF_WEIGHTS_NAME + ".index")
+                tf2_weights_2_path = get_local_path(TF2_WEIGHTS_NAME)
+                flax_weights_path = get_local_path(FLAX_WEIGHTS_NAME)
+                safetensors_weights_path = get_local_path(SAFE_WEIGHTS_NAME) + variant_suffix
+                safetensors_weights_index_path = get_local_path(SAFE_WEIGHTS_INDEX_NAME) + variant_suffix
+                pytorch_weights_path = get_local_path(WEIGHTS_NAME) + variant_suffix
+                pytorch_weights_index_path = get_local_path(WEIGHTS_INDEX_NAME) + variant_suffix
```

Comment:
Let's make a small util that compute the name with the variant so we don't have to add this code and can just replace all names below by (for instance)
```
_add_variant(WEIGHTS_NAME, variant)
```